.circleci/config.yml

```diff
@@ -1105,19 +1106,21 @@ jobs:
   # Builds in C++20 mode and uses debug build in order to speed up.
   # Do *NOT* store any artifacts or workspace as we don't run tests on this build.
   b_ubu_cxx20:
-    <<: *base_ubuntu2404_xlarge
+    <<: *base_ubuntu2404_large
     environment:
-      <<: *base_ubuntu2404_xlarge_env
+      <<: *base_ubuntu2404_large_env
       CMAKE_BUILD_TYPE: Debug
-      CMAKE_OPTIONS: -DCMAKE_CXX_STANDARD=20
-      MAKEFLAGS: -j 10
+      CMAKE_OPTIONS: -DCMAKE_CXX_STANDARD=20 -DUSE_LD=gold
+      MAKEFLAGS: -j 6
```

Comment:
We could just leave this as `-j` and it should automatically select the appropriate number of threads/processes based on the core count.

Reply:
That effectively means `-j 4` and leads to significantly longer runtime due to lower resource utilization. It's a trade-off of course, overdoing it (looking at you, `-j 10` on a 4 vCPU resource) might lead to OOM.